collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java

```diff
@@ -222,16 +221,15 @@ public void onClick(DialogInterface dialog, int i) {
     @Override
     public Loader<Cursor> onCreateLoader(int id, Bundle args) {
         showProgressBar();
-        return new FormsDao().getFormsCursorLoader(getFilterText(), getSortingOrder());
+
+        boolean uniqueByFormId = GeneralSharedPreferences.getInstance().getBoolean(PreferenceKeys.KEY_HIDE_OLD_FORM_VERSIONS, false);
+        return new FormsDao().getFormsCursorLoader(getFilterText(), getSortingOrder(), uniqueByFormId);
     }
 
     @Override
     public void onLoadFinished(@NonNull Loader<Cursor> loader, Cursor cursor) {
         hideProgressBarIfAllowed();
-        listAdapter.changeCursor(
-                GeneralSharedPreferences.getInstance().getBoolean(PreferenceKeys.KEY_HIDE_OLD_FORM_VERSIONS, false)
-                        ? FormUtils.removeOldForms(cursor)
-                        : cursor);
+        listAdapter.swapCursor(cursor);
```

Comment:
The [documentation for `onLoadFinished`](https://developer.android.com/reference/android/support/v4/app/LoaderManager.LoaderCallbacks#onLoadFinished(android.support.v4.content.Loader%3CD%3E,%20D)) says: "if the Cursor is being placed in a CursorAdapter, you should use the swapCursor(android.database.Cursor) method so that the old Cursor is not closed."

Was there a particular reason to use `changeCursor` instead?

Reply:
I have no idea what was the reason.